lib/std/start.zig

```diff
@@ -437,20 +437,22 @@ fn posixCallMainAndExit() callconv(.C) noreturn {
             std.os.linux.pie.relocate(phdrs);
         }
 
-        // ARMv6 targets (and earlier) have no support for TLS in hardware.
-        // FIXME: Elide the check for targets >= ARMv7 when the target feature API
-        // becomes less verbose (and more usable).
-        if (comptime native_arch.isARM()) {
-            if (at_hwcap & std.os.linux.HWCAP.TLS == 0) {
-                // FIXME: Make __aeabi_read_tp call the kernel helper kuser_get_tls
-                // For the time being use a simple abort instead of a @panic call to
-                // keep the binary bloat under control.
-                std.os.abort();
+        if (builtin.thread_local_storage) {
```

Comment:
```suggestion
        if (builtin.thread_local_storage and !builtin.single_threaded) {
```
> It shaves off a bunch of bytes* for single-threaded programs that don't need TLS

So shouldn't it be like this then? I think there's still a need for the `thread_local_storage` flag though because even a multithreaded application may not need TLS, I think.